src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\ErrorHandler\Debug;
+use Symfony\Component\ErrorHandler\ErrorHandler;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\TerminableInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SymfonyRuntime implements RuntimeInterface
+{
+    private $request;
+    private $input;
+    private $output;
+    private $application;
+    private $command;
+
+    public function __construct(string $projectDir = null)
+    {
+        $projectDir = $projectDir ?? \dirname(__DIR__, 3);
+
+        if (!class_exists(Dotenv::class)) {
```

Comment:
Should this really be a hard requirement? 

I suggest only loading `.env` if the component exists.

Reply:
I think it should yes, that's part of the default Symfony experience now: first-class env var support, dev tooling included.

Reply:
But it is not uncommon that you've defined all your environment variables as environment variables without using the Dotenv. 

Does this component do something more that I'm missing?

Reply:
We've had this discussion on recipes already: we concluded that `.env` files are best always loaded, at least to provided defaults.
And we built `composer dump-env` to make this ultra fast on prod.

Reply:
Could it be moved into `index.php` or `bin/console` - I always had an option to remove that from sample files which are delivered framework so I always did cause it's not common you have `.env` file on production.
All the time it was a developer choice.

Reply:
What about dockerized apps? They use envs provided on start. You suggest running `composer dump-env` in production image each time on start? for what purpose?

Reply:
It is actually very common to load `.env` files in production since this is the default configuration and what we recommend doing.

> You suggest running composer dump-env in production image each time on start?

What is suggested currently (unrelated to this PR) is running this command yes - not on start but when creating the image. See e.g. https://github.com/api-platform/api-platform/blob/master/api/Dockerfile#L89 when NOT loading `.env` files is desired.

Reply:
@nicolas-grekas I still can't understand the "why?".. is there something behind it?

Reply:
> what we recommend doing
there is a little difference of `recommend doing` and `force users to do`. We adopted sf5 recently and we still using good old `.env local + .env.dist` approach since it's aligned with all other existing projects. Having runtime to fail is forcing users to adopt new `.env` schema

For our case it's impossible to create `prod` image `.env` file while building. We store all production settings in separate storage which is used only when deploying to production. We will not build different images for different deployment locations

Reply:
@gmponos because it makes the DX consistent and closes a DX trap, eg. #https://github.com/symfony/recipes/issues/497. Since we've made this change, things just work and we have no report anymore, while we had too many before- proof this provides the most expected behavior.

Reply:
@scaytrase in any doubt, the current way of doing things still works and will always continue to work, forever, because that's code in *your* hands. 

Later on, if you want to adopt the proposed way of running your app and the `SymfonyRuntime` doesn't fit your needs, you'll have your own implementation of `RuntimeInterface` (which could compose the parts of `SymfonyRuntime` that fit).

> We store all production settings in separate storage

That's not incompatible: hybrid (real + dotenv) env vars is the norm in Symfony apps. `.env` files contain only the defaults - they can also be empty.

Reply:
@nicolas-grekas I still don't get it why it cannot stay outside of `SymfonyRuntime` class and be left there where it should be which IMO is https://github.com/symfony/recipes/blob/master/symfony/console/5.1/bin/console#L31 and it was there all the time so developers had a choice of removing it. I get that it is recommended. But with that, I'm not forced to use it, so I can easily remove it without loading `.env` and w/o consequences run my app. With this change IMHO you move too much responsibility from invoked script to that class for the price of enforcing users to use `.env` or overriding the constructor.

Reply:
@brzuchal I have a proposal, PR updated so you can have a look: `SymfonyRuntime` now takes an array of options. The `dotenv` one can be set to `false` to bypass load `.env` files. I also added a new `$_SERVER['APP_RUNTIME_OPTIONS']` to let the `autoload.php` script know which option to pass when instantiating `SymfonyRuntime`.

This means your front-controller would start with:
```php
$_SERVER['APP_RUNTIME_OPTIONS'] = ['dotenv' => false];
require_once __DIR__.'/vendor/symfony/runtime/autoload.php';
```

Reply:
Looks much more flexible now! Thank you

Reply:
@nicolas-grekas That looks really nice and flexible now. Thx!